sdk/identity/identity/samples/AzureIdentityExamples.md

```diff
@@ -908,6 +915,185 @@ class RotatingCertificateCredential implements TokenCredential {
+
+#### Persisting the token cache
+
+The credential handles persisting all the data needed to silently authenticate one or many accounts. It manages sensitive data such as refresh tokens and access tokens which must be protected to prevent compromising the accounts related to them. By default, the `@azure/identity` library will protect and cache sensitive token data using available platform data protection.
+
+To configure a credential, such as the `InteractiveBrowserCredential`, to persist token data, simply set the `tokenCachePersistenceOptions` option.
+
+```ts
+import { useIdentityExtension, InteractiveBrowserCredential } from "@azure/identity";
+import { cachePersistenceExtension } from "@azure/identity-cache-persistence";
+
+useIdentityExtension(cachePersistenceExtension);
+
+const credential = new InteractiveBrowserCredential({
+  tokenCachePersistenceOptions: {
+    enabled: true
+  }
+});
+```
+
+#### Persisting the Authentication Record
+
+The `AuthenticationRecord` which is returned from the `authenticate`, contains data identifying an authenticated account. It is needed to identify the appropriate entry in the persisted token cache to silently authenticate on subsequent executions. There is no sensitive data in the `AuthenticationRecord` so it can be persisted in a non-protected state.
+
+Here is an example of an application storing the `AuthenticationRecord` to the local file system after authenticating the user.
+
+```ts
+import fs from "fs";
+import { promisify } from "util";
+import path from "path";
+const AUTH_RECORD_PATH = "./tokencache.bin";
+
+const authRecord: AuthenticationRecord = await credential.authenticate();
+const writeFileAsync = promisify(fs.writeFile);
+const content = serializeAuthenticationRecord(authRecord);
+await writeFileAsync(path.join(process.cwd(), AUTH_RECORD_PATH), content);
+```
+
+#### Silent authentication with Authentication Record and Token Cache Persistence Options
+
+Once an application has configured a credential to persist token data and an `AuthenticationRecord`, it is possible to silently authenticate. This example demonstrates an application setting the `tokenCachePersistenceOptions` and retrieving an `AuthenticationRecord` from the local file system to create an `InteractiveBrowserCredential` capable of silent authentication.
+
+```ts
+import { useIdentityExtension, InteractiveBrowserCredential } from "@azure/identity";
+import { cachePersistenceExtension } from "@azure/identity-cache-persistence";
+
+useIdentityExtension(cachePersistenceExtension);
+
+const AUTH_RECORD_PATH = "./tokencache.bin";
+const readFileAsync = promisify(fs.readFile);
+const fileContent = await readFileAsync(path.join(process.cwd(), AUTH_RECORD_PATH), {
```

Comment:
remember to wrap this in a `async function main() {` to ensure it is easier for developers to copy, paste and run this

Reply:
done